### PR TITLE
Implement appmodule.AppModule interface for ibc-go packages' modules

### PR DESCRIPTION
**Pull request by pysel**
closes: #4217

## Description

This PR implements `appmodule.AppModule` interface for ibc-go modules (except `capability`, it was already implemented). 

Also introduces some cleanups related to basic manager.

### CODE REVIEW

**Comment by faddat** [comment]
thank you!

**Review by crodriguezvega — COMMENTED** [review]
I think this is looking good, @pysel! Thank you for the contribution. Is there anything you still need to do to make the PR ready for review?

**Comment by pysel** [comment]
hey @crodriguezvega! yeah, looks good, just not sure why it is not passing one of ci checks (one of Tests)

**Comment by DimitrisJim** [comment]
> just not sure why it is not passing one of ci checks (one of Tests)

Fixed the linting issues present, the other one (building for amd) also fails in target branch, there's a PR for e2es that should fix that.

Thanks again for this!

**Review by crodriguezvega — APPROVED** [review]
Great work, @pysel! And thanks a bunch for making the linter happy, @DimitrisJim! :)

File: modules/apps/27-interchain-accounts/module.go [hunk 1]
```diff
@@ -46,6 +49,12 @@ func (AppModuleBasic) Name() string {
 	return types.ModuleName
 }
 
+// IsOnePerModuleType implements the depinject.OnePerModuleType interface.
+func (AppModule) IsOnePerModuleType() {}
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/module.go:53]: any idea what does this API infer to the depinject/runtime setup?

**Review by damiannolan — APPROVED** [review]
Awesome, thank you! 🙏

**Review by DimitrisJim — APPROVED** [review]

File: modules/apps/27-interchain-accounts/module.go (hunk 1 above)
Comment (DimitrisJim) [reply modules/apps/27-interchain-accounts/module.go:53]: https://docs.cosmos.network/main/architecture/adr-057-app-wiring#dependency-injection touches on it *slightly*, haven't had time to read further into it tho

**Pull request merged**